src/darwin/Framework/CHIP/templates/MTRClusterNames.zapt

```diff
@@ -0,0 +1,12 @@
+{{> header excludeZapComment=true}}
+
+#import <Matter/MTRClusterConstants.h>
+#import <Matter/MTRDefines.h>
+#import <stdint.h>
+
+NS_ASSUME_NONNULL_BEGIN
+@interface MTRClusterAttributeIDTypeToString : NSObject
+- (NSString *)formatMTRClusterIDTypeToString:(MTRClusterIDType)clusterID;
+- (NSString *)formatMTRAttributeIDTypeToString:(MTRClusterIDType)clusterID :(MTRAttributeIDType)attributeID;
+@end
```

Comment:
It's funny; I was just being asked for API for this today.  So thank you for doing this!

The proposed API shape I was looking at was to just have free functions, something like:
```
NSString * MTRClusterNameForID(MTRClusterIDType clusterID) MTR_NEWLY_AVAILABLE;
NSString * MTRAttributeNameForID(MTRClusterIDType clusterID, MTRAttributeIDType attributeID) MTR_NEWLY_AVAILABLE;
```

That seems better than hanging these off a new random interface, probably, but does it play nicely enough with Swift?

Past that:

1. Whatever we add here has to have MTR_NEWLY_AVAILABLE annotations.
2. In terms of the implementation, a walk over a data table does seem better than a switch, at least for cluster names, if it's not too hard to do.
3. There needs to be API documentation, especially because the behavior for "unknown ID" is not at all obvious.
4. For the attribute name case, we have run into issues with huge generated functions that include all the attributes, which we have handled by having a switch on cluster followed by calls to separate functions per-cluster.  That might not be an issue with the data table approach, though.

Reply:
> Regarding https://github.com/project-chip/connectedhomeip/pull/31026 

That's a nice coincidence.

> That seems better than hanging these off a new random interface, probably, but does it play nicely enough with Swift?

I am using it like this in a Swift app:
```swift
let c2s = MTRClusterAttributeIDType()
c2s.formatMTRAttributeIDType(
       MTRClusterIDType(rawValue: UInt32(truncating: cluster))!,
       MTRAttributeIDType(rawValue: UInt32(truncating: attribute))! )
```
and it worked nicely also for huge clusters.

I am open to all proposed changes, but I am a ObjectiveC/Swift noobie.

Reply:
Yeah, I just checked and that works fine from Swift.  So I would propose:
```
MTR_EXPORT MTR_NEWLY_AVAILABLE NSString * MTRClusterNameForID(MTRClusterIDType clusterID);
MTR_EXPORT MTR_NEWLY_AVAILABLE NSString * MTRAttributeNameForID(MTRClusterIDType clusterID, MTRAttributeIDType attributeID);
```